api/libproject/gcloudcontext.rb

```diff
@@ -0,0 +1,36 @@
+require_relative "../../libproject/utils/common"
+require_relative "environments"
+require "json"
+
+class GcloudContextV2
+  attr_reader :account, :project
+
+  def initialize(env_key)
+    # We use both gcloud and gsutil commands for various tasks. While gcloud can take arguments,
+    # gsutil uses the current gcloud config, so we just ensure a reasonable config is present here
+    # rather than allowing arguments.
+    common = Common.new
+    common.status "Reading glcoud configuration..."
+    configs = common.capture_stdout \
+        %W{docker-compose run --rm api gcloud --format=json config configurations list}
+    active_config = JSON.parse(configs).select{|x| x["is_active"]}.first
+    common.status "Using '#{active_config["name"]}' gcloud configuration"
+    @account = active_config["properties"]["core"]["account"]
+    @project = active_config["properties"]["core"]["project"]
+    common.status "  account: #{@account}"
+    common.status "  project: #{@project}"
+    unless @account && @project
+      common.error "Account and project required for gcloud config." \
+          " See gcloud config configurations --help."
+      exit 1
+    end
+    unless @account.end_with?("@pmi-ops.org")
+      common.error "Account is not a pmi-ops.org account: #{@account}"
+      exit 1
+    end
+    unless @project == Environments::PROJECT_NAMES[env_key]
+      common.error "Project '#{@project}' does not match current environment key #{env_key}"
+      exit 1
+    end
```

Comment:
I think this approach is safer than requiring a `--project` argument. I also think it is simpler to grab configuration from one location (your gcloud config) rather than allowing an override here (since your gcloud config will likely affect things anyway).

Reply:
If you don't use --project when you run commands for messing with a cloud environment, it is very easy to accidentally update the wrong one. I have done this on multiple occasions using gcloud app deploy. It can break things pretty badly.

So I think using --project on each command should absolutely be required.

Reply:
Are you assuming here we're going to have different Ruby commands for different cloud environments, passing in different values of env_key?

That doesn't scale. We are going to have five different cloud environments -- we don't want to maintain five different versions of each command. We can just use --project with a single set of commands to avoid this.